### PR TITLE
switch to using json sanitizer for admin counters, gauges, and text-readouts

### PR DESCRIPTION
**Pull request by jmarantz**
Commit Message: Uses the new Json::sanitizer from #20637 to generate counters, gauges, and text-readouts in response to admin endpoint `/stats?format=json`, resulting in a 5.6x speedup for AllCountersJson, and making JSON output only around 10% slower than text output.

Additional Description:
Before:
```
------------------------------------------------------------------
Benchmark                        Time             CPU   Iterations
------------------------------------------------------------------
BM_AllCountersText             454 ms          453 ms            2
BM_UsedCountersText           35.8 ms         35.8 ms           20
BM_FilteredCountersText       1806 ms         1805 ms            1
BM_AllCountersJson            2824 ms         2823 ms            1
BM_UsedCountersJson           61.8 ms         61.8 ms           11
BM_FilteredCountersJson       1798 ms         1798 ms            1
```
after
```
------------------------------------------------------------------
Benchmark                        Time             CPU   Iterations
------------------------------------------------------------------
BM_AllCountersText             458 ms          458 ms            2
BM_UsedCountersText           36.1 ms         36.1 ms           19
BM_FilteredCountersText       1790 ms         1790 ms            1
BM_AllCountersJson             496 ms          496 ms            2
BM_UsedCountersJson           36.4 ms         36.4 ms           19
BM_FilteredCountersJson       1789 ms         1789 ms            1
```

Risk Level: low -- Json::sanitizer is new, but based on nlohmann, and is differentially fuzzed vs protobuf serialization
Testing: //test/server/admin/...
Docs Changes: n/a
Release Notes: n/a
Platform Specific Features: n/a

### CODE REVIEW

File: source/server/admin/stats_render.h [hunk 1]
```diff
@@ -96,13 +79,14 @@ class StatsJsonRender : public StatsRender {
                       const std::vector<uint64_t>& interval_buckets,
                       const std::vector<uint64_t>& cumulative_buckets);
 
-  std::vector<ProtobufWkt::Value> stats_array_;
   ProtobufWkt::Struct histograms_obj_;
   ProtobufWkt::Struct histograms_obj_container_;
   std::unique_ptr<ProtobufWkt::ListValue> histogram_array_;
   bool found_used_histogram_{false};
-  bool first_{true};
+  absl::string_view delim_{""};
```
Comment (wbpcode) [inline source/server/admin/stats_render.h:86]: nit: `absl::string_view delim_;`

**Review by wbpcode — COMMENTED** [review]
LGTM overall. Only one nit comment.

File: source/server/admin/stats_render.h (hunk 1 above)
Comment (wbpcode) [reply source/server/admin/stats_render.h:86]: get it.

**Review by wbpcode — APPROVED** [review]
LGTM.

**Review by zuercher — APPROVED** [review]

**Comment by zuercher** [comment]
@jmarantz I'll leave the merge to you

**Pull request merged**